WordPress/Classes/ViewRelated/Notifications/Views/ListTableViewCell+Notifications.swift

```diff
@@ -0,0 +1,75 @@
+/// Encapsulates logic that configures `ListTableViewCell` with `Notification` models.
+///
+extension ListTableViewCell {
+    /// Configures the cell based on the provided `Notification` object.
+    func configureWithNotification(_ notification: Notification) {
+        // indicator view
+        indicatorColor = Style.unreadIndicatorColor
+        showsIndicator = !notification.read
+
+        // avatar image
+        configureImage(with: notification.iconURL)
+
+        // title text
+        attributedTitleText = notification.renderSubject()
+
+        // snippet text
+        snippetText = notification.renderSnippet()?.string ?? String()
+    }
+
+    /// Toggles an overlay view that presents an undo option for the user.
+    /// Whether the view is displayed or not depends on the `text` parameter.
+    /// Given an empty text, the method will try to dismiss the overlay text (if it exists).
+    /// Otherwise, it will try to show the overlay view on top of the cell.
+    /// - Parameters:
+    ///     - text: The text describing the action that can be undone.
+    ///     - onUndelete: The closure to be executed when the undo button is tapped.
+    func configureUndeleteOverlay(with text: String?, onUndelete: @escaping () -> Void) {
+        guard let someText = text, !someText.isEmpty else {
+            dismissOverlay()
+            return
+        }
+
+        let undoOverlayView = makeUndoOverlayView()
+        undoOverlayView.textLabel.text = text
+        undoOverlayView.actionButton.on(.touchUpInside) { _ in
+            onUndelete()
+        }
+
+        showOverlay(with: undoOverlayView)
+    }
+
+    // MARK: Private Helpers
+
+    /// Creates a pre-styled overlay view based on `ListSimpleOverlayView`.
+    /// This will be used to show options for the user to revert the action performed on the notification.
+    private func makeUndoOverlayView() -> ListSimpleOverlayView {
```

Comment:
Just a thought - you have a `private extension`, this could go in there as well, keeping all the `private`s together.

Reply:
Done in https://github.com/wordpress-mobile/WordPress-iOS/pull/16885/commits/c2e5665e291b16065b9edb5d2fba013277428dde 🙏